### PR TITLE
Urls to docs in rust_hir

### PR DESCRIPTION
**Pull request by sancho20021**

### CODE REVIEW

File: compiler/rustc_hir/src/hir.rs [hunk 1]
```diff
@@ -1632,6 +1632,10 @@ pub struct ConstBlock {
 }
 
 /// An expression.
+///
+/// For more details, see the [rust lang reference]
```
Comment (compiler-errors) [inline compiler/rustc_hir/src/hir.rs:1636]:
```suggestion
/// For more details, see the [rust lang reference].
```

File: compiler/rustc_hir/src/hir.rs [hunk 2]
```diff
@@ -3146,6 +3150,10 @@ impl ItemId {
 /// An item
 ///
 /// The name might be a dummy name in case of anonymous items
+///
+/// For more details, see the [rust lang reference]
```
Comment (compiler-errors) [inline compiler/rustc_hir/src/hir.rs:3154]:
```suggestion
/// For more details, see the [rust lang reference].
```

**Review by compiler-errors — CHANGES_REQUESTED** [review]
Please squash this into one commit as well.

I also wonder if we should move this comment to `ItemKind` and `ExprKind`. Also it may be useful to note that the reference does not document nightly-only features, and that there may be slight differences in the names and representation of AST nodes between the compiler and the reference.

File: compiler/rustc_hir/src/hir.rs (hunk 1 above)
Comment (sancho20021) [reply compiler/rustc_hir/src/hir.rs]: fixed

File: compiler/rustc_hir/src/hir.rs (hunk 2 above)
Comment (sancho20021) [reply compiler/rustc_hir/src/hir.rs]: fixed

**Comment by sancho20021** [comment]
> I also wonder if we should move this comment to `ItemKind` and `ExprKind`.
Maybe, but I see `Item` and `Expr` as the main entities which the developer operates with in many functions. Just less clicks to find the reference.

> Also it may be useful to note that the reference does not document nightly-only features, and that there may be slight differences in the names and representation of AST nodes between the compiler and the reference.
Left a note

**Review by compiler-errors — APPROVED** [review]

**Pull request merged**